app/controllers/ping/application_controller.rb

```diff
@@ -1,7 +1,26 @@
 module Ping
   class ApplicationController < ActionController::API
+    FORCE_LOGGING = !!ENV.fetch('FORCE_LOG_PING', false)
+
     def ping
       render plain: "PONG"
     end
+
+    def process(action, *args)
+      if self.class.force_log_ping?
+        super
+      else
+        logger.silence do
+          super
+        end
+      end
+    end
+
+    private
```

Comment:
`private` will not affect class methods. Should use `private_class_method` like http://ruby-doc.org/core-2.5.3/Module.html#method-i-private_class_method

Reply:
Thanks!
`Ping::ApplicationController.stub :force_log_ping?, true` does not work with a private method